homeassistant/components/recorder/statistics.py

```diff
@@ -1373,6 +1373,71 @@ def _get_newest_sum_statistic_in_sub_period(
     return newest_sum
 
 
+def resolve_period(
```

Comment:
Do we have standalone tests for this ? 

Should we use TypedDict to type `period_def` ? 

Since this is not perse related to statistics, should we move this to a util/dt.py or something similar?

Reply:
(if moved to util, it should raise `ValueError` instead of `HomeAssistantError` as util methods are not aware of HA.

Reply:
This has been addressed, but the helper is moved to `components/recorder/util.py` instead of to `util/dt.py`